src/base/emu-i386/kvm.c

```diff
@@ -361,13 +361,13 @@ static int init_kvm_vcpu(void)
      the kernel needs to emulate that using V86 mode, as is necessary
      on Nehalem and earlier Intel CPUs */
   ret = ioctl(vmfd, KVM_SET_TSS_ADDR,
-	      (unsigned long)DOSADDR_REL(monitor->kvm_tss));
+	      sregs.tr.base + offsetof(struct monitor, kvm_tss));
   if (ret == -1) {
     perror("KVM: KVM_SET_TSS_ADDR\n");
     return 0;
   }
 
-  uint64_t addr = DOSADDR_REL(monitor->kvm_identity_map);
+  uint64_t addr = sregs.tr.base + offsetof(struct monitor, kvm_identity_map);
   ret = ioctl(vmfd, KVM_SET_IDENTITY_MAP_ADDR, &addr);
```

Comment:
Seems you need to checck
KVM_CAP_SET_IDENTITY_MAP_ADDR
before using that ioctl.
I am not really sure what it does though...

Reply:
For older Intel CPUs it needs a page table so KVM can boot in V86 mode (it can't boot in real mode there), so it's the same reason as KVM_SET_TSS_ADDR. We boot directly in V86 mode anyway so this may not be needed but it doesn't hurt (and dmesg may complain if you don't set those).

Reply:
So why not to check for cap
KVM_CAP_SET_IDENTITY_MAP_ADDR?

Reply:
Done in commit 3fd4f4a92c